### PR TITLE
Add `crmUrl function` to smarty in the standard way

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Even with Smarty2 this method of registering functions was being phased out for putting the file in the folder

### CODE REVIEW

File: CRM/Core/Smarty/plugins/function.crmURL.php [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+/**
+ * Generate a url
+ *
+ * @param string $string
+ *   The html to be tweaked with.
+ *
+ * @return string
+ */
+function smarty_function_crmURL($string, $params) {
+  return CRM_Utils_System::url();
```
Comment (totten) [inline CRM/Core/Smarty/plugins/function.crmURL.php:28]: If that passes, it says more about the inadequacy of our test suite...

Comment (eileenmcnaughton) [reply CRM/Core/Smarty/plugins/function.crmURL.php:28]: yeah I didn't want to overdo it

**Comment by totten** [comment]
> Even with Smarty2 this method of registering functions was being phased out for putting the file in the folder

This confused me at first, but I think I see what you're saying. Within _CiviCRM's usage of Smarty v2_, the typical code-pattern is to split these things apart as separate plugin-files -- which makes it a bit nicer for browsing/discovering. (*The Smarty project isn't pushing for the change -- but it's a matter of our local convention.*)

Just note that `CRM_Utils_System::crmURL()` and `CRM_Utils_System::url()` have very different signatures, and there are a couple other callers for `crmURL()`. (Not many - I didn't see any in the broader `universe`. So it's amenable to change.)

**Comment by eileenmcnaughton** [comment]
@totten I can't say as I love the signature of our php function 


image

**Comment by totten** [comment]
lol, yeah, changing `crmURL()` is a lot easier than changing `url()`...

**Comment by eileenmcnaughton** [comment]
This is what the usages look like

```{capture assign=icalFeed}{crmURL p='civicrm/event/ical' q="reset=1&id=`$event.id`" h=0 a=1 fe=1}{/capture}```

- `p` & `q` are drupal conventions  - no idea what they stand for but I'm kinda used to them being 'the first bit & the second bit' of the url

No idea about `a` & `fe` `h`

**Comment by eileenmcnaughton** [comment]
Ah so `crmUrl` converts those 'handy letters' into parameters .....

image

**Comment by totten** [comment]
On the test-site, I spot-checked a few screens that use `{crmURL}` -- all worked fine. Merging.

**Pull request merged**